### PR TITLE
refactor(components): Rename Memory Component for Better Clarity and Consistency

### PR DESCRIPTION
**Pull request by Vigtu**
# Changes in Helper Components:

## Component Renames:
1. Memory -> MessageRetriever:
- Renamed class from MemoryComponent to MessageRetrieverComponent
- Updated display name to "Message Retriever"
- Changed icon to "message-square-text"
- Updated component name to "MessageRetriever"

## Import Updates:
- Updated import statements in __init__.py
- Renamed imported components:
  * MemoryComponent -> MessageRetrieverComponent

## File Renames:
- memory.py -> message_retriever.py

## No Functional Changes:
- Core functionality remains the same
- Only naming and cosmetic changes applied
- Component descriptions maintained

### CODE REVIEW

**Review by ogabrielluiz — CHANGES_REQUESTED** [review]
Same for this one. We need to keep the name as it was otherwise the update function might not work.

File: src/backend/base/langflow/components/helpers/message_store.py [hunk 1]
```diff
@@ -7,11 +7,11 @@
...
 
-class StoreMessageComponent(Component):
-    display_name = "Store Message"
+class MessageStoreComponent(Component):
+    display_name = "Message Store"
     description = "Stores a chat message or text into Langflow tables or an external memory."
-    icon = "save"
-    name = "StoreMessage"
+    icon = "message-square-text"
+    name = "MessageStore"
```
Comment (ogabrielluiz) [inline src/backend/base/langflow/components/helpers/message_store.py:14]: This name can't be changed either

**Review by ogabrielluiz — CHANGES_REQUESTED** [review]

File: src/backend/base/langflow/components/helpers/message_store.py (hunk 1 above)
Comment (Vigtu) [reply src/backend/base/langflow/components/helpers/message_store.py]: Got it. Is it a problem to change the Component Class as well? (MessageStoreComponent)

Comment (ogabrielluiz) [reply src/backend/base/langflow/components/helpers/message_store.py]: If you set the `name` it ignores the class name. By default `__class__.__name__` is used. Only older components require us to set `name` for backwards compatibility.

**Review by ogabrielluiz — APPROVED** [review]

**Comment by Vigtu** [comment]
I believe that the changes are no longer necessary, due to the implementations that are already being made in the memory component, that said I will be closing this PR

**Pull request closed without merging**